server.go

```diff
@@ -26,6 +26,11 @@ var (
 	defaultTimeout     = 10 * time.Second
 )
 
+type ConnectionStateChange struct {
+	Identifier string
+	Connected  bool
```

Comment:
It used to be enum but I finally decided that 2 value enum is a little bit overdone.

:+1: I'll check what can I do for error propagation